src/content/app/genome-browser/components/track-panel/components/track-panel-list/track-panel-items/TrackPanelGene.tsx

```diff
@@ -63,67 +66,76 @@ const TrackPanelGene = (props: TrackPanelGeneProps) => {
     getBrowserTrackState(state, {
       genomeId,
       objectId: focusObjectId,
-      tracksGroup: 'objectTracks',
-      categoryName: 'main',
-      trackId: GENE_TRACK_ID
+      tracksGroup: 'objectTracks'
     })
   );
-  const { toggleTrack, genomeBrowser } = useGenomeBrowser();
-
+  const visibleTranscriptIds = useAppSelector(getVisibleTranscriptIds);
+  const { toggleTrack, updateTranscriptTracks, genomeBrowser } =
+    useGenomeBrowser();
   const dispatch = useAppDispatch();
 
+  let allTranscriptsInGene: TrackPanelTranscriptType[] | undefined; // eslint-disable-line
```

Comment:
why is eslint disabled for this line?

Reply:
Because it gets reassigned at line number 138 and eslint keeps complaining of not reassigning it.

Reply:
Why don't we assign an empty array to both `allTranscriptsInGene` and `sortedTranscripts`? Then you wouldn't need to run checks on `allTranscriptsInGene` to make sure that it exists.

Reply:
This was something we did previously. However, that caused these variables to get reset all the time, which resulted in `useEffect` getting executed multiple times: https://github.com/Ensembl/ensembl-client/pull/771#discussion_r909306155

Reply:
`const allTranscriptsInGene = currentData?.gene.transcripts ?? []`?

Reply:
That did the trick! ✅

Reply:
Why does it need to be reassigned? I can also see the reassignment on line 136, which says ` allTranscriptsInGene = gene.transcripts;`; but on line 77 you are already assigning this value to the `allTranscriptsInGene` variable.

Reply:
Yes that's true. Will remove line 136.